packages/modals-collection/src/docker/add-docker-app-to-homarr.tsx

```diff
@@ -0,0 +1,90 @@
+import { Button, Group, Image, List, LoadingOverlay, Stack, Text, TextInput } from "@mantine/core";
+
+import type { RouterOutputs } from "@homarr/api";
+import { clientApi } from "@homarr/api/client";
+import { useForm, zodResolver } from "@homarr/form";
+import { createModal } from "@homarr/modals";
+import { useI18n } from "@homarr/translation/client";
+import { z } from "@homarr/validation";
+
+interface AddDockerAppToHomarrProps {
+  selectedContainers: RouterOutputs["docker"]["getContainers"]["containers"];
+}
+
+export const AddDockerAppToHomarr = createModal<AddDockerAppToHomarrProps>(({ actions, innerProps }) => {
+  const t = useI18n();
+  const form = useForm({
+    validate: zodResolver(
+      z.object({
+        containerUrls: z.array(z.string().url().nullable()),
+      }),
+    ),
+    initialValues: {
+      containerUrls: innerProps.selectedContainers.map((container) => {
+        if (container.ports[0]) {
+          return `http://${container.ports[0].IP}:${container.ports[0].PublicPort}`;
+        }
+
+        return null;
+      }),
+    },
+  });
+  const { mutate, isPending } = clientApi.app.createMany.useMutation({
+    onSuccess() {
+      actions.closeModal();
+    },
+  });
+  const handleSubmit = () => {
+    mutate(
+      innerProps.selectedContainers.map((container, index) => ({
+        name: container.name,
+        iconUrl: container.iconUrl,
+        description: null,
+        href: form.values.containerUrls[index] ?? null,
+      })),
+    );
+  };
+  return (
+    <form onSubmit={form.onSubmit(handleSubmit)}>
+      <LoadingOverlay visible={isPending} zIndex={1000} overlayProps={{ radius: "sm", blur: 2 }} />
+      <Stack>
+        <List>
+          {innerProps.selectedContainers.map((container, index) => (
+            <List.Item
+              styles={{ itemWrapper: { width: "100%" }, itemLabel: { flex: 1 } }}
+              icon={<Image src={container.iconUrl} alt="container image" w={30} h={30} />}
+              key={container.id}
+            >
+              <Group justify="space-between">
+                <Text>{container.name}</Text>
+                <TextInput
+                  onChange={(event) => {
+                    form.setFieldValue("containerUrls", (prev) => {
+                      prev[index] = event.target.value.length > 0 ? event.target.value : null;
+                      return prev;
+                    });
+                    form.validate();
+                  }}
+                  defaultValue={form.values.containerUrls[index] ?? ""}
```

Comment:
This sould probably instead by value= and you should be able to use form.getInputProps(`containerUrls.${index}`)